From 8beddb2fa64ebd471535b2cbab7ec04b3cf7b53a Mon Sep 17 00:00:00 2001
From: Benedikt Ritter <britter@apache.org>
Date: Sun, 24 Feb 2013 11:43:32 +0000
Subject: [PATCH] [BEANUTILS-426] - ConstructorUtils.invokeConstructor(Class
 klass, Object arg) throws NullPointerException when arg==null, now also fixed
 for invokeExactConstrucot(Class, Object)

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/beanutils/trunk@1449460 13f79535-47bb-0310-9956-ffa450edef68
---
 .../java/org/apache/commons/beanutils/ConstructorUtils.java | 3 +++
 .../apache/commons/beanutils/ConstructorUtilsTestCase.java  | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/src/main/java/org/apache/commons/beanutils/ConstructorUtils.java b/src/main/java/org/apache/commons/beanutils/ConstructorUtils.java
index daca3c188..897a48661 100644
--- a/src/main/java/org/apache/commons/beanutils/ConstructorUtils.java
+++ b/src/main/java/org/apache/commons/beanutils/ConstructorUtils.java
@@ -194,6 +194,9 @@ public static Object invokeExactConstructor(Class klass, Object arg)
             InstantiationException {
 
         Object[] args = { arg };
+        if (arg == null) {
+            args = null;
+        }
         return invokeExactConstructor(klass, args);
     }
 
diff --git a/src/test/java/org/apache/commons/beanutils/ConstructorUtilsTestCase.java b/src/test/java/org/apache/commons/beanutils/ConstructorUtilsTestCase.java
index 375cf7da3..70ccb030a 100644
--- a/src/test/java/org/apache/commons/beanutils/ConstructorUtilsTestCase.java
+++ b/src/test/java/org/apache/commons/beanutils/ConstructorUtilsTestCase.java
@@ -147,6 +147,12 @@ public void testInvokeExactConstructor() throws Exception {
         }
     }
 
+    public void testInvokeExactConstructorWithNull() throws Exception {
+        Object obj = ConstructorUtils.invokeExactConstructor(TestBean.class, (Object) null);
+        assertNotNull(obj);
+        assertTrue(obj instanceof TestBean);
+    }
+
     public void testInvokeExactConstructorWithArgArray() throws Exception {
         {
             Object[] args = { new Float(17.3f), "TEST" };
